app/src/main/java/org/mozilla/fenix/utils/StartupTaskManager.kt

```diff
@@ -0,0 +1,40 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.utils
+
+typealias StartupTask = () -> Unit
+
+/**
+ * A queue of tasks that are performed at specific points during Fenix startup.
+ *
+ * This queue contains a list of startup tasks. Each task in the queue will be started once Fenix
+ * is visually complete (as long as it has not be cancelled since it was added to the queue).
+ * Each task will execute on the main thread.
+ *
+ * This is threadsafe.
+ */
+class StartupTaskManager {
+    private var tasks = mutableListOf<StartupTask>()
+
+    /**
+     * Add a task to the queue.
+     *
+     * @param task: The task to add to the queue.
+     */
+    @Synchronized
+    private fun add(task: StartupTask) {
+        tasks.add(task)
+    }
+
+    /**
+     * Start all tasks in the queue. When all the tasks have been started,
+     * clear the queue.
+     */
+    @Synchronized
+    private fun start() {
+        tasks.forEach { it.run {} }
+        tasks.clear()
```

Comment:
Why do we need to clear the queue?

One thought: anything captured by the lambda will remain captured if we hold on to these tasks, taking up more memory than we need to. If this is the reason for this to exist, we should comment so.